opr/functions.py

```diff
@@ -137,21 +137,22 @@ def e260_ssnn_calc(sequence):
         e260 -= BASE_EXTINCTION_COEFFICIENTS[base]
     return e260
 
-
-def protein_seq_calc(rna_sequence, frame):
+def protein_seq_calc(rna_sequence, frame=1, one_letter=True):
     """
     Calculate the sequence of amino acids from a mRNA sequence.
 
     :param rna_sequence: mRNA sequence
     :type rna_sequence: str
     :param frame: frame of translation (1, 2, or 3)
     :type frame: int
+    :param one_letter: Whether to return amino acids in 1-letter codes (True) or 3-letter codes (False).
+    :type one_letter: bool
     :return: protein sequence
     """
-    frame = frame - 1
-    protein_seq = ""
-    for i in range(frame, len(rna_sequence), 3):
-        codon = rna_sequence[i:i + 3]
-        if len(codon) == 3:
-            protein_seq += CODONS_TO_AMINO_ACIDS[codon]
-    return protein_seq
+    start = frame - 1
+    protein = []
+    for i in range(start, len(rna_sequence) - 2, 3):
+        codon = rna_sequence[i:i+3]
+        aa1 = CODONS_TO_AMINO_ACIDS_AA1.get(codon, 'X')
+        protein.append(AA1_TO_AA3[aa1] if not one_letter else aa1)
+    return ''.join(protein) if one_letter else '-'.join(protein)
```

Comment:
If we have a `CODONS_TO_AMINO_ACIDS_AA3`, you can have a `codon_to_acid` dictionary which is assigned based on the `one_letter` (or `multi_letter)` parameter that is more robust and easy to read.